scripts/minikube.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash -e
+
+#https://github.com/rook/rook/blob/master/tests/scripts/minikube.sh
+
+function wait_for_ssh() {
+    local tries=100
+    while ((tries > 0)); do
+        if minikube ssh echo connected &>/dev/null; then
+            return 0
+        fi
+        tries=$((tries - 1))
+        sleep 0.1
+    done
+    echo ERROR: ssh did not come up >&2
+    exit 1
+}
+
+function copy_image_to_cluster() {
+    local build_image=$1
+    local final_image=$2
+    if [[ "${VM_DRIVER}" == "none" ]]; then
+        docker pull "${build_image}"
+        docker tag "${build_image}" "${final_image}"
+        return
+    fi
+    docker save "${build_image}" | (eval "$(minikube docker-env --shell bash)" && docker load && docker tag "${build_image}" "${final_image}")
+}
+
+# configure minikube
+MINIKUBE_VERSION=${MINIKUBE_VERSION:-"latest"}
+KUBE_VERSION=${KUBE_VERSION:-"v1.13.0"}
+MEMORY=${MEMORY:-"3000"}
+VM_DRIVER=${VM_DRIVER:-"virtualbox"}
+#configure image repo
+CEPHCSI_IMAGE_REPO=${CEPHCSI_IMAGE_REPO:-"quay.io/cephcsi"}
+K8S_IMAGE_REPO=${K8S_IMAGE_REPO:-"quay.io/k8scsi"}
+
+#feature-gates for kube
+K8S_FEATURE_GATES=${K8S_FEATURE_GATES:-"BlockVolume=true,CSIBlockVolume=true,VolumeSnapshotDataSource=true"}
+# install minikube
+function install_minikube() {
+    if type minikube >/dev/null 2>&1; then
+        local version
+        version="$(minikube version)"
+        echo "minikube already installed with ${version}"
+        return
+    fi
+
+    echo "Installing minikube. Version: ${MINIKUBE_VERSION}"
+    curl -Lo minikube https://storage.googleapis.com/minikube/releases/"${MINIKUBE_VERSION}"/minikube-linux-amd64 && chmod +x minikube && mv minikube /usr/local/bin/
+}
+
+function install_kubectl() {
+    # Download kubectl, which is a requirement for using minikube.
+    echo "Installing kubectl. Version: ${KUBE_VERSION}"
+    curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/"${KUBE_VERSION}"/bin/linux/amd64/kubectl && chmod +x kubectl && mv kubectl /usr/local/bin/
+}
+
+case "${1:-}" in
+up)
+    install_minikube
+    #if driver  is none install kubectl with KUBE_VERSION
+    if [[ "${VM_DRIVER}" == "none" ]]; then
+        mkdir -p "$HOME"/.kube "$HOME"/.minikube
+        install_kubectl
+    fi
+
+    echo "starting minikube with kubeadm bootstrapper"
+    minikube start --memory="${MEMORY}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --vm-driver="${VM_DRIVER}" --feature-gates="${K8S_FEATURE_GATES}"
+
+    DISK="sda1"
+    if [[ "${VM_DRIVER}" == "kvm2" ]]; then
+        # use vda1 instead of sda1 when running with the libvirt driver
+        DISK="vda1"
+    fi
+    # create a link so the default dataDirHostPath will work for this
+    # environment
+    if [[ "${VM_DRIVER}" != "none" ]]; then
+        wait_for_ssh
+        # shellcheck disable=SC2086
+        minikube ssh "sudo mkdir -p /mnt/${DISK}/${PWD}; sudo mkdir -p $(dirname $PWD); sudo ln -s /mnt/${DISK}/${PWD} $(dirname $PWD)/"
```

Comment:
(query) I am not sure why this is required, any pointers?

Reply:
https://github.com/rook/rook/commit/24d7c3059b8f5cf8fc92ce7659b31ebe6a151953

Reply:
I did see the link above, the question was not around why DISK needed to be special across the virtualization environments (which is what the commit rectifies). The question is why the soft links and all, or rather what does this mean "create a link so the default dataDirHostPath will work for this environment" does for Rook, do we know?

Reply:
This comment is more a query, not something blocking the merge BTW.